etf/decode.go

```diff
@@ -438,7 +438,9 @@ func Decode(packet []byte, cache []Atom) (retTerm Term, retByte []byte, retErr e
 
 			b := make([]byte, n)
 			copy(b, packet[5:n+5])
-			b[n-1] = b[n-1] >> (8 - bits)
+			if bits != 8 {
+				b[n-1] = b[n-1] >> (8 - bits)
+			}
```

Comment:
useless

Reply:
Benchmark data first before coming to a conclusion?